todo_app/data/session_items.py

```diff
@@ -67,3 +94,27 @@ def save_item(item):
     session['items'] = updated_items
 
     return item
+
+def delete_item(id):
+    """
+    Delete an existing item in the session. If no existing item matches the ID of the specified item, nothing is deleted.
+
+    Args:
+        item: The item to delete.
+    """
+    items = get_items()
+    indexNumber = find_index(items, 'id', id)
+    del items[indexNumber]
+    #updated_items = [item if item['id'] == existing_item['id'] else existing_item for existing_item in existing_items]
+
+    session['items'] = items
+   
+    return items
+
+def find_index(dicts, key, value):
+    class Null: pass
+    for i, d in enumerate(dicts):
+        if d.get(key, Null) == value:
+            return i
```

Comment:
The `Null` class definition is totally unnecessary.

You can just do d.get(key)

Python has `None` for situations like this but that's already the default result for `get` when you don't specify your own.